util/logtools/split_log.js

```diff
@@ -314,25 +282,25 @@ const writeFile = (outputFile, startLine, endLine) => {
 
   const collector = await makeCollectorFromPrepass(logFileName);
 
-  if (args['search_fights'] === null) {
+  if (args['search_fights'] === -1) {
```

Comment:
For my own understanding, why did all the nulls change to -1?

Reply:
Because when used in combination with `nargs = '?'`, a `defaultValue` will always be passed to the consuming script even if the flag was not actually present at the command line:

```
node --loader ts-node/esm split_log -f log.log -lf 1
// console.log(JSON.stringify(args));
{"file":"log.log","force":null,"search_fights":1,"search_zones":-1,"fight_regex":-1,"zone_regex":-1}

node --loader ts-node/esm split_log -f log.log -lf -lz
// console.log(JSON.stringify(args));
{"file":"log.log","force":null,"search_fights":null,"search_zones":null,"fight_regex":-1,"zone_regex":-1}
```

It Seems To Me (and I recognize that it's opinion) that it doesn't make a lot of sense to have nulls indicate that a flag may actually *be* present but without a value present at the command line.

Conversely, if we set `constant` instead, the default value will be passed to the consuming script if and only if the flag is present. A null value in this case means explicitly that the flag was not used, while a default value explicitly means "a magic number is present, handle it accordingly".

```
 node --loader ts-node/esm split_log -f log.log -lf
// console.log(JSON.stringify(args));
{"file":"log.log","timeline":null,"force":null,"search_fights":-1,"search_zones":null,"fight_regex":null,"zone_regex":null,"adjust":null}
```

Reply:
Thanks for the explanation! I get it now.